File: fieldsync-inventory-backend/src/main/java/com/example/fieldsync_inventory_api/dto/inventory/InventoryBatchResponseDTO.java
package com.example.fieldsync_inventory_api.dto.inventory;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class InventoryBatchResponseDTO {
    private BigDecimal totalStock;
    private BigDecimal totalGraded;
    private BigDecimal totalUngraded;
    private BigDecimal totalGerminated;
    private BigDecimal totalUngerminated;
    private List<InventoryBatchDTO> batches;
}

File: fieldsync-inventory-app/app/src/main/java/com/example/fieldsync_inventory_app/data/remote/dto/inventory/InventoryBatchResponseDto.kt
package com.example.fieldsync_inventory_app.data.remote.dto.inventory

import com.google.gson.annotations.SerializedName

data class InventoryBatchResponseDto(
    @SerializedName("totalStock")
    val totalStock: Double,
    @SerializedName("totalGraded")
    val totalGraded: Double,
    @SerializedName("totalUngraded")
    val totalUngraded: Double,
    @SerializedName("totalGerminated")
    val totalGerminated: Double,
    @SerializedName("totalUngerminated")
    val totalUngerminated: Double,
    @SerializedName("batches")
    val batches: List<BatchStockDto>
)

data class BatchStockDto(
    @SerializedName("batchId")
    val batchId: Int,
    @SerializedName("varietyId")
    val varietyId: Int,
    @SerializedName("varietyName")
    val varietyName: String,
    @SerializedName("year")
    val year: Int,
    @SerializedName("seasonId")
    val seasonId: Int,
    @SerializedName("seasonName")
    val seasonName: String,
    @SerializedName("seasonDescription")
    val seasonDescription: String,
    @SerializedName("generationId")
    val generationId: Int,
    @SerializedName("generationName")
    val generationName: String,
    @SerializedName("grading")
    val grading: Boolean,
    @SerializedName("germination")
    val germination: Boolean,
    @SerializedName("stock")
    val stock: Double
)
